src/upgrade/sender.rs

```diff
@@ -1,28 +1,26 @@
-use async_std::sync;
-
 use crate::upgrade::Connection;
 
 /// The sending half of a channel to send an upgraded connection.
 ///
-/// Unlike `async_std::sync::channel` the `send` method on this type can only be
+/// Unlike `async_channel::Sender` the `send` method on this type can only be
 /// called once, and cannot be cloned. That's because only a single instance of
 /// `Connection` should be created.
 #[derive(Debug)]
 pub struct Sender {
-    sender: sync::Sender<Connection>,
+    sender: async_channel::Sender<Connection>,
 }
 
 impl Sender {
     /// Create a new instance of `Sender`.
     #[doc(hidden)]
-    pub fn new(sender: sync::Sender<Connection>) -> Self {
+    pub fn new(sender: async_channel::Sender<Connection>) -> Self {
```

Comment:
Is this a breaking change? This may be doc-hidden but if async-h1 depends on it there could be problems.

Reply:
I don't believe `async-h1` depends on it ([search results](https://github.com/http-rs/async-h1/search?q=Sender%3A%3Anew&unscoped_q=Sender%3A%3Anew)); if I recall correctly this is only exposed so that we can use it in integration tests. You're right though that if we have another crate depending on this it would complicate things..

Reply:
This does seem to be a breaking change. I don't why anyone would use `Sender::new` since there's no way to add that sender to a `Request`/`Response` type, but 🤷. It's also gated behind the "unstable" feature, and I couldn't find any dependent crates that enable this.

Maybe it's still a good idea to wait for the next breaking release?

Reply:
I would also suggest making this function private, since it has no real use in the public API.

Reply:
I believe only the upgrade sender stuff is behind `"unstable"`, were it all we could do that change with less worry.

Reply:
Only `upgrade::Sender::new` is a breaking change right? Nothing outside "unstable" should be affected.

Reply:
Just noticed `trailers::Sender::new` and `trailers::Receiver::new` are outside "unstable" and they're affected too. But I don't think anyone would use those functions outside of `http-types`; they're completely useless unless they're in a `Request`/`Response` type.

Reply:
both `trailers::Sender::new` and `upgrade::Sender::new` are non-public methods; neither of these affects our stability